skimage/transform/_thin_plate_splines.py

```diff
@@ -0,0 +1,299 @@
+        return radial_dist
+
+    def _spline_function(self, x, y, radial_dist, coeffs):
+        """Solve the spline function in the X and Y directions"""
+        w = coeffs[:-3]
+        a1, ax, ay = coeffs[-3:]
+        summation = np.dot(w, radial_dist)
+        return a1 + ax * x + ay * y + summation
+
+
+def _radial_basis_kernel(r):
+    """Compute basis kernel for thine-plate splines.
+
+    Parameters
+    ----------
+    r : (4, N) ndarray
+        Input array representing the norm distance between interlandmark
+        distances for the source form and based on the (x,y) coordinates
+        for each of these points.
+
+    Returns
+    -------
+    U : (4, N) ndarray
+        Calculated kernel function U.
+    """
+    _small = 1e-8  # Small value to avoid divide-by-zero
+    r_sq = r**2
+    U = np.where(r == 0.0, 0.0, r_sq * np.log(r_sq + _small))
+    return U
+
+
+def tps_warp(
+    image, src, dst, *, output_region=None, interpolation_order=1, grid_scaling=1
+):
+    """Return an array of warped images.
+
+    Define a thin-plate-spline warping transform that warps from the
+    src to the dst, and then warp the given images by
+    that transform.
+
+    Parameters
+    ----------
+    image : ndarray
+        Input image.
+    src : (N, 2)
+        Control point at source coordinates.
+    dst : (N, 2)
+        Control point at target coordinates.
+    output_region : tuple of integers, optional
+        The region ``(xmin, ymin, xmax, ymax)`` of the output
+        image that should be produced. (Note: The region is inclusive, i.e.
+        xmin <= x <= xmax)
+    interpolation_order : int, optional
+        If value is 1, then use linear interpolation else use
+        nearest-neighbor interpolation.
+    grid_scaling : int, optional
+        If grid_scaling is greater than 1, say x, then the transform is
+        defined on a grid x times smaller than the output image region.
+        Then the transform is bilinearly interpolated to the larger region.
+        This is fairly accurate for values up to 10 or so.
+
+    Returns
+    -------
+    warped : array_like
+        The warped input image.
+
+
+    References
+    ----------
+    .. [1] Bookstein, Fred L. "Principal warps: Thin-plate splines and the
+           decomposition of deformations," IEEE Transactions on pattern analysis
+           and machine intelligence 11.6 (1989): 567–585.
+
+    Examples
+    --------
+    Produce a warped image rotated by 90 degrees counter-clockwise.
+
+    >>> import skimage as ski
+    >>> astronaut = ski.data.astronaut()
+    >>> image = ski.color.rgb2gray(astronaut)
+    >>> src = np.array([[0, 0], [0, 512], [512, 512],[512, 0]])
+    >>> dst = np.array([[512, 0], [0, 0], [0, 512],[512, 512]])
+    >>> output_region = (0, 0, image.shape[0], image.shape[1])
+    >>> tform = ski.transform.TpsTransform()
+    >>> tform.estimate(src, dst)
+    True
+    >>> warped_image = ski.transform.tps_warp(
```

Comment:
Why doesn't `tform` get used in what follows?

Reply:
Good catch. I guess this is a relic of the previous API. 

```python
    >>> tform = ski.transform.TpsTransform()
    >>> tform.estimate(src, dst)
    True
```
can probably be removed.